processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanAggregatorFactory.java

```diff
@@ -0,0 +1,180 @@
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.aggregation.Aggregator;
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.AggregatorUtil;
+import org.apache.druid.query.aggregation.BufferAggregator;
+import org.apache.druid.query.aggregation.VectorAggregator;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.vector.VectorColumnSelectorFactory;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ */
+public class DoubleMeanAggregatorFactory extends AggregatorFactory
+{
+  private final String name;
+  private final String fieldName;
+
+  @JsonCreator
+  public DoubleMeanAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") final String fieldName
+  )
+  {
+    this.name = Preconditions.checkNotNull(name, "null name");
+    this.fieldName = Preconditions.checkNotNull(fieldName, "null fieldName");
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return "doubleMean";
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return DoubleMeanHolder.MAX_INTERMEDIATE_SIZE;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    return new DoubleMeanAggregator(metricFactory.makeColumnValueSelector(fieldName));
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    return new DoubleMeanBufferAggregator(metricFactory.makeColumnValueSelector(fieldName));
+  }
+
+  @Override
+  public VectorAggregator factorizeVector(final VectorColumnSelectorFactory selectorFactory)
+  {
+    return new DoubleMeanVectorAggregator(selectorFactory.makeValueSelector(fieldName));
+  }
+
+  @Override
+  public boolean canVectorize()
+  {
+    return true;
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return DoubleMeanHolder.COMPARATOR;
+  }
+
+  @Nullable
+  @Override
+  public Object combine(@Nullable Object lhs, @Nullable Object rhs)
+  {
+    if (lhs instanceof DoubleMeanHolder && rhs instanceof DoubleMeanHolder) {
+      return ((DoubleMeanHolder) lhs).update((DoubleMeanHolder) rhs);
+    } else {
+      throw new IAE(
+          "lhs[%s] or rhs[%s] not of type [%s]",
+          lhs.getClass().getName(),
+          rhs.getClass().getName(),
```

Comment:
The case of null value should be considered.

Reply:
they can't be null because doubleMean aggregator impls never return null.

Reply:
Sorry, I mean `lhs` or `rhs` may be null, which causes `getClass()` to throw NPE.

Reply:
thanks, added the null handling.